starters/spring/src/main/kotlin/br/com/zup/beagle/spring/filter/BeaglePlatformFilter.kt

```diff
@@ -29,20 +29,32 @@ import javax.servlet.http.HttpServletResponse
 
 class BeaglePlatformFilter(private val objectMapper: ObjectMapper) : Filter {
 
-    override fun doFilter(request: ServletRequest?, response: ServletResponse?, chain: FilterChain?) {
-        if (chain != null && request is HttpServletRequest && response is HttpServletResponse) {
-            request.setAttribute(
-                BeaglePlatformUtil.BEAGLE_PLATFORM_HEADER,
-                request.getHeader(BeaglePlatformUtil.BEAGLE_PLATFORM_HEADER)
-            )
-            val responseWrapper = ContentCachingResponseWrapper(response)
+    override fun doFilter(
+        request: ServletRequest?,
+        response: ServletResponse?,
+        chain: FilterChain?
+    ) {
+        val isServletRequestResponse = request is HttpServletRequest && response is HttpServletResponse
+        val httpServletResponse = response as? HttpServletResponse
+        val httpServletRequest = request as? HttpServletRequest
+        val platformHeader =
+            if (isServletRequestResponse)
+                httpServletRequest?.getHeader(BeaglePlatformUtil.BEAGLE_PLATFORM_HEADER)
+            else null
+
+        if (chain != null && isServletRequestResponse && platformHeader != null) {
+            request?.setAttribute(BeaglePlatformUtil.BEAGLE_PLATFORM_HEADER, platformHeader)
+            val responseWrapper = ContentCachingResponseWrapper(httpServletResponse)
             chain.doFilter(request, responseWrapper)
-            treatResponse(responseWrapper, request.getHeader(BeaglePlatformUtil.BEAGLE_PLATFORM_HEADER))
+            treatResponse(responseWrapper, platformHeader)
             responseWrapper.copyBodyToResponse()
         }
```

Comment:
The filter will skip the request like this, it should go continue down the chain if the checks fail.
```suggestion
        } else {
            chain?.doFilter(request, response)
        }
```